### PR TITLE
Release Manager plugin

### PR DESCRIPTION
**Pull request by johanbaath**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by johanbaath** [comment]
This is a work in progress!

**Review by miixell — APPROVED** [review]

File: plugins/release-manager/src/ReleaseManagerPage.test.tsx
Comment (Rugvip) [inline plugins/release-manager/src/ReleaseManagerPage.test.tsx:23]: Use this pattern instead for 98% fewer headaches: https://github.com/backstage/backstage/blob/7647449390e9167c6dee247ce67d570bb7181761/packages/test-utils/src/testUtils/appWrappers.test.tsx#L108 😅 

In general there should be very few reasons to use jest module mocks. The backstage is is structured to allow for more direct injection patterns, and encourages those patterns within plugins as well.

File: yarn.lock
Comment (Rugvip) [inline yarn.lock:14]: Let's avoid a full `yarn.lock` refresh here. Please reset to the one in master and just include the ones needed by the plugin.

Comment (johanbaath) [reply yarn.lock:14]: Absolutely!

File: plugins/release-manager/src/ReleaseManagerPage.test.tsx
Comment (johanbaath) [reply plugins/release-manager/src/ReleaseManagerPage.test.tsx:23]: Yeah, the tests haven't been updated for a while. We have been migrating the plugin to keep up with the changes over a couple of months. Thanks for the example, will take care of it 👍 !

**Comment by EDjur** [comment]
@Rugvip Feel free to review again the latest changes :)

File: packages/create-app/templates/default-app/packages/app/src/sidebar.tsx
Comment (Rugvip) [inline packages/create-app/templates/default-app/packages/app/src/sidebar.tsx:36]: I think we should hold off on adding this to the template for now. It also requires a few more changes than this 😅

File: plugins/release-manager/src/ReleaseManagerPage.test.tsx
Comment (Rugvip) [inline plugins/release-manager/src/ReleaseManagerPage.test.tsx:35]: We tend to use `msw` over this, but just leaving that as a suggestion and not a blocker

File: plugins/release-manager/src/ReleaseManagerPage.test.tsx
Comment (Rugvip) [inline plugins/release-manager/src/ReleaseManagerPage.test.tsx:58]: Just use this instead :grin:

```ts
new ConfigReader({
  releasemanager: {
    android: {
      baseUrl: '',
      identifier: 'com.spotify.music',
    },
  },
})
```

File: plugins/release-manager/src/components/AndroidReleaseNotesPane.tsx
Comment (Rugvip) [inline plugins/release-manager/src/components/AndroidReleaseNotesPane.tsx:31]: We avoid using `FC`, see https://backstage.io/docs/architecture-decisions/adrs-adr006

File: plugins/release-manager/package.json
Comment (Rugvip) [inline plugins/release-manager/package.json:47]: We (ben) worked hard to remove this, use `msw` instead 😅 

See https://backstage.io/docs/architecture-decisions/adrs-adr007

**Comment by adamdmharvey** [comment]
If you don't mind the question, I assume this PR supports #659 so just tagging it.  The screen shots in the linked issue look really cool and I can absolutely see the value this could bring to the community at large!

Query on the name.  When you say _Release Manager_, is this some sort of Android-specific tool relating to publishing only Android apps?

I only ask as the wording "Release Manager" is super generic, and say if I referred to the DevOps Periodic Table of Tools, I might think it could be any one of these:

image

My team also has home grown tools to do this sort of thing.  So as a conceptual way to manage releases, understand the latest out there, I totally get it, but wondering if this should be a bit more specific as it's not a generic release manager plugin that could be used by anyone, but something like a Mobile Release Manager or maybe even specifically Android Release Manager? (and thinking forward to when this is released, adding it to the [Plugin Marketplace](https://backstage.io/plugins), how is it pitched).

Hope you don't mind the question!

**Comment by benjdlambert** [comment]
Hey @johanbaath is there any update on the progress of this? How long do we expect this to be in a draft state?

**Comment by johanbaath** [comment]
> Hey @johanbaath is there any update on the progress of this? How long do we expect this to be in a draft state?

Hey! Not sure. I just switched teams at Spotify and have no time allocated for this at the moment. @EDjur maybe you can provide a better answer?

**Comment by EDjur** [comment]
Hi @benjdlambert . I need to raise internally how to prioritise this. Will get back to you. :)

**Comment by benjdlambert** [comment]
@EDjur no worries!

**Comment by regicsolutions** [comment]
We have a need for a Release Management Plugin that wouldn't be just specific to mobile releases but to all types of releases (app, microservice, mobile) and act as a plugin that would only be used by a Release Manager where they could control the version and dependencies of releases moving from QA to Production. Our initial thinking was to have a segregated repo that is locked down to the release manager called app-release with its own CI file and deployment.properties file that would basically hold a 3 things:

```
namespace: my-namespace
version: 1.0.9
services: fronend-service, backend-service

```

Cant seem to find any documentation on this plugin, anyone know if its specific to android/mobile releases?

**Comment by EDjur** [comment]
@benjdlambert Currently this isn't super high prio from our end unfortunately. We have a ticket in the backlog to get around to it eventually. I can perhaps have a go next week to address some of the comments in here :)

**Comment by regicsolutions** [comment]
@adamdmharvey interesting article on release management: https://medium.com/ynap-tech/beyond-gitops-how-we-release-our-microservices-on-kubernetes-at-ynap-683617cfd3cc 

YNAP built a homegrown tool for release management called Slipway which is exactly what we are looking for to solve our release management gaps. unfortunately, it was never open-sourced but wanted to share as maybe it can be used for some potential inspiration, would love to get your thoughts! 

2A6EAEEB-A97A-49E6-BA33-6A858EB72C64
FCA43FC1-B2FF-495F-9798-993A63209609
EB91803B-7DA3-4A73-BBB1-951B54C010D7

**Comment by EDjur** [comment]
@adamdmharvey 
> Query on the name. When you say _Release Manager_, is this some sort of Android-specific tool relating to publishing only Android apps?

I agree the naming might need another look. Internally, we use this as a frontend for various backend services that collate data regarding the release process of our main android and ios music apps. Meaning that in and by itself, the plugin doesn't do *that* much - it requires a backend to provide data.

The general idea here is that we've open sourced [android-store-service](https://github.com/spotify/android-store-service) as a baseline backend. This service is essentially a stripped down version of what we run internally (we removed some Spotify-specific stuff).

> I only ask as the wording "Release Manager" is super generic, and say if I referred to the DevOps Periodic Table of Tools, I might think it could be any one of these:
> 
> My team also has home grown tools to do this sort of thing. So as a conceptual way to manage releases, understand the latest out there, I totally get it, but wondering if this should be a bit more specific as it's not a generic release manager plugin that could be used by anyone, but something like a Mobile Release Manager or maybe even specifically Android Release Manager? (and thinking forward to when this is released, adding it to the [Plugin Marketplace](https://backstage.io/plugins), how is it pitched).
> 
> Hope you don't mind the question!

Not at all - I appreciate the question! And totally agree. Mobile Release Manager strikes the balance right in my opinion. Because as I said, internally we use this for both ios/android.

**Comment by EDjur** [comment]
@regicsolutions I apologise if it wasn't super clear in the initial PR, but this release-manager (or rather *mobile-release-manager*) plugin is intended for only mobile releases. 

Correct me if I'm understanding your screenshots the wrong way, but that looks more akin to what the Deployment view and Kubernetes integrations in Backstage allow you to do (or at least parts of it) :)

**Comment by regicsolutions** [comment]
@EDjur thanks I think mobile release manager makes sense. I am looking for something that is micro-service's oriented. The Kubernetes plugin is infrastructure focused.

**Pull request closed without merging**
